mobile/lib/main.dart

```diff
@@ -560,12 +561,39 @@ Future<void> configureFirebase(FlutterLocalNotificationsPlugin localNotification
   messaging.subscribeToTopic('all');
 }
 
-/// Ask the user for permission to send notifications via Firebase
-Future<void> requestNotificationPermission() async {
+Future<void> ensureStoringFirebaseToken() async {
   FirebaseMessaging messaging = FirebaseMessaging.instance;
-  final token = await messaging.getToken();
+  messaging.getToken().then((token) {
+    if (token != null) {
+      FLog.info(text: "Firebase token: $token");
+      updateFcmToken(token).then((_) {
+        FLog.info(text: "Firebase token updated");
+      });
+    } else {
+      FLog.warning(text: "Firebase token is null");
+    }
+  });
+
+  // Firebase sometimes updates tokens at runtime, make sure we handle that case
+  messaging.onTokenRefresh.listen((String token) {
```

Comment:
Ah, so it is not final, then I would not put it into the registration request

Reply:
yes, it's not final; it can change at any arbitrary point in time (it doesn't change often though).